app/src/main/java/ch/epfl/sweng/favors/location/AndroidGeocoder.java

```diff
@@ -0,0 +1,32 @@
+package ch.epfl.sweng.favors.location;
+
+import android.content.Context;
+import android.location.Address;
+import android.location.Geocoder;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+
+public class AndroidGeocoder extends GeocoderDispatcher {
+    private static AndroidGeocoder androidGeocoder;
+    private Geocoder geocoder;
+
+    private AndroidGeocoder(Context context){
+        geocoder = new Geocoder(context, Locale.getDefault());
+    }
+
+    public static AndroidGeocoder getGeocoder(Context context) {
+        return new AndroidGeocoder(context);
```

Comment:
I prefered to use a factory than a singleton pattern. I made this choice because I do not see the need to have only one Geocoder since Geocoder is not a singleton in the android API.

Reply:
and an important thing is that a geocoder need a context, so a singleton is not a good idea in this case.

Reply:
Okay thank your for the explanation